apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts

```diff
@@ -183,12 +183,16 @@ export class UpsertWorkflowUseCase {
   }
 
   private async upsertWorkflowPreferences(workflow: NotificationTemplateEntity, command: UpsertWorkflowCommand) {
+    if (!command.workflowDto.preferences?.workflow) {
```

Comment:
This DTO looks wider than it should. The comment is not actionable as part of this PR, but we must make it wider rather than deeper. Let's treat it as a code smell.

Reading this code, the hypothesis is that this check ensures that the workflow has preferences, right?

Reply:
correct, we ensure that we have workflow preferences to upsert.